Leorik.Core/Evaluation/Evaluation.cs

```diff
@@ -50,8 +53,19 @@ internal void Update(BoardState board, ref Move move)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void UpdateScore(BoardState board)
         {
-            float score = EvalBase() + NormalizePhase(PhaseValue) * EvalEndgame();
-            Score = (short)(Endgame.IsDrawn(board) ? (int)score >> 4 : score);
+            //Todo - this is too slow - cache or seperate quiesce updatescore?
+            //Research requried: maybe this is called already only in quiet positions.
+            //NNue only valid when NOT in check. 
+            if (!board.InCheck())

```

Comment:
Question for Thomas

Reply:
BoardState provides the static evaluation of a position and the search is making sure that the static eval is only returned for leaf nodes that are actually quiet. So the InCheck tests shouldn't be necessary.